libr/io/map.c

```diff
@@ -139,8 +140,41 @@ void io_map_calculate_skyline(RIO *io) {
 			last_map = map;
 		}
 	}
-
 	r_binheap_clear (&heap);
+
+	const RPVector *skyline = &io->map_skyline;
+	RPVector *shadow = &io->map_skyline_shadow;
+	RInterval *cur_itv = NULL;
+
+	for (i = 0; i < r_pvector_len (skyline); i++) {
+		const RIOMapSkyline *part = r_pvector_at (skyline, i);
+		ut64 part_from = part->itv.addr;
+		ut64 part_size = part->itv.size;
+
+		if (!cur_itv) {
+			cur_itv = R_NEW (RInterval);
```

Comment:
null deref. better use r_itv_new()  (which doesnt exist yet. but feel free to add it)

Reply:
like: cur_itv = r_itv_new (part_from, part_size);